src/backend/distributed/utils/metadata_cache.c

```diff
@@ -677,6 +722,32 @@ InvalidateDistRelationCacheCallback(Datum argument, Oid relationId)
 			cacheEntry->isValid = false;
 		}
 	}
+
+	/* now check whether we're in the middle of dropping the extension */
+
+	/* if we're not in a transaction then we'll crash while looking up oids */
+	inTransaction = IsTransactionState();
+	if (!inTransaction)
+	{
+		return;
+	}
+
+	shardPlacementId = DistShardPlacementRelationId();
+
+	elog(WARNING,"Invalidate rel: %d target: %d", relationId, shardPlacementId);
+
+	if (relationId != InvalidOid && relationId == shardPlacementId)
+	{
+		shardPlacementId = get_relname_relid("pg_dist_shard_placement",
```

Comment:
Invalidation events seem rare enough that we could just drop the cache if there is an invalidation event on a metadata table. I tried checking relationId == metadataCacheState.distPartitionRelationId and it seems that the only place where we get an invalidation message for pg_dist_partition in the regression tests is when we create and drop the extension.